examples/lib/stories/effects/remove_effect_example.dart

```diff
@@ -19,18 +18,22 @@ class RemoveEffectExample extends FlameGame with HasTappableComponents {
     camera.viewport = FixedResolutionViewport(Vector2(400, 600));
     final rng = Random();
     for (var i = 0; i < 20; i++) {
-      add(_RandomCircle(rng));
+      add(_RandomCircle.random(rng));
     }
   }
 }
 
 class _RandomCircle extends CircleComponent with Tappable {
-  _RandomCircle(Random rng) : super(radius: rng.nextDouble() * 30 + 10) {
-    position.setValues(
+  factory _RandomCircle.random(Random rng) {
+    final radius = rng.nextDouble() * 30 + 10;
+    final position = Vector2(
       rng.nextDouble() * 320 + 40,
       rng.nextDouble() * 520 + 40,
     );
-    paint.color = Colors.primaries[rng.nextInt(Colors.primaries.length)];
+    final paint = Paint()
+      ..color = Colors.primaries[rng.nextInt(Colors.primaries.length)];
+    return CircleComponent(radius, position: position, paint: paint)
+        as _RandomCircle;
```

Comment:
This is super neat, but now when I run the "Remove example" I get the following error:
```
======== Exception caught by widgets library =======================================================
The following TypeErrorImpl was thrown building FutureBuilder<void>(dirty, state: _FutureBuilderState<void>#3a6be):
Expected a value of type '_RandomCircle', but got one of type 'CircleComponent'

When the exception was thrown, this was the stack: 
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 251:49  throw_
packages/flame/src/game/game_widget/game_widget.dart 319:25                   <fn>
packages/flutter/src/widgets/async.dart 782:48                                build
packages/flutter/src/widgets/framework.dart 4782:27                           build
packages/flutter/src/widgets/framework.dart 4665:15                           performRebuild
packages/flutter/src/widgets/framework.dart 4840:11                           performRebuild
packages/flutter/src/widgets/framework.dart 4355:5                            rebuild
packages/flutter/src/widgets/framework.dart 2620:31                           buildScope
packages/flutter/src/widgets/binding.dart 882:9                               drawFrame
packages/flutter/src/rendering/binding.dart 319:5                             [_handlePersistentFrameCallback]
packages/flutter/src/scheduler/binding.dart 1143:15                           [_invokeFrameCallback]
packages/flutter/src/scheduler/binding.dart 1080:9                            handleDrawFrame
packages/flutter/src/scheduler/binding.dart 996:5                             [_handleDrawFrame]
lib/_engine/engine/platform_dispatcher.dart 1003:13                           invoke
lib/_engine/engine/platform_dispatcher.dart 157:5                             invokeOnDrawFrame
lib/_engine/engine.dart 440:45                                                <fn>
====================================================================================================
```

Reply:
Ah thought I tried that, it must have been running the old code, added a default constructor.
I found a problem with the events not coming in now, so fixing that.

Reply:
Now it should be working, had some problems with the anchor code.